src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -174,11 +178,68 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     this.setTransform(device, nodes);
   }
 
+  private async chooseVideoTransformInputDevice(device: VideoTransformDevice): Promise<void> {
+    if (this.chosenVideoTransformDevice === device) {
+      this.logger.info('Used same video transform device');
+      return;
+    }
+
+    this.chosenVideoTransformDevice = device;
+    const inner: Device = await device.intrinsicDevice();
+
+    // check if the current video device can be reused.
+    if (!this.isMediaStreamReusableByDeviceId(this.activeDevices['video']?.stream, inner)) {
+      this.logger.info('video transform device needs new intrinsic device');
+      await this.chooseInputIntrinsicDevice('video', inner, false, true);
+    } else {
+      // When saved stream is reusable, only switch the saved stream to filtered stream for sending
+      // but keep the saved stream intact.
+
+      // Note: to keep the chosen media stream intact, it is important to avoid a full stop
+      // because videoTileUpdate can be called when video is stopped and user might call `bindVideoElement` to disconnect the element.
+      // In current implementation, disconnecting the element will `hard` stop the media stream.
+
+      const newMediaStream = this.activeDevices['video'].stream;
+
+      // Input is not a MediaStream.
+      if (!(inner as MediaStream).id) {
+        await newMediaStream.getVideoTracks()[0].applyConstraints(inner as MediaTrackConstraints);
+      }
+
+      // `applyProcessors` will start processing.
+      await device.applyProcessors(this.activeDevices['video'].stream);
+
+      if (this.boundAudioVideoController?.videoTileController.hasStartedLocalVideoTile()) {
+        // optimized method exists, a negotiation can be avoided
+        if (!!this.boundAudioVideoController.replaceLocalVideo) {
```

Comment:
This is always defined on `DAVC`, and it will throw for Plan B clients, right?

Reply:
right, https://github.com/aws/amazon-chime-sdk-js/pull/875#discussion_r527092694